src/store/sagas/subjectSaga.ts

```diff
@@ -0,0 +1,40 @@
+import { call, put, takeEvery } from 'redux-saga/effects';
+import showToast from 'utils/toastMessage';
+import { fetchSubjects, createSubject } from 'api/subject';
+import { SUBJECTS_API, CREATE_SUBJECT_API } from 'store/types';
+import { 
+  fetchSubjectsFulfilled, fetchSubjectsRejected, 
+  createSubjectFulfilled, createSubjectRejected, 
+  fetchSubjectsStarted, createSubjectStarted
+} from 'store/reducers/subjectReducer';
+import { IApiResponse } from 'utils/types';
+
+function* fetchSubjectsSaga() {
+  try{
+    yield put(fetchSubjectsStarted());
```

Comment:
you don't need to call started action separately same action can be listened in reducer and saga as well.